Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/DocumentObj/SetIshDocumentObj.cs

```diff
@@ -148,122 +239,135 @@ protected override void ProcessRecord()
-                    else
+
+                    var returnFields = IshObject[0].IshFields ?? new IshFields();
+                    IshFields requestedMetadata = IshSession.IshTypeFieldSetup.ToIshRequestedMetadataFields(IshSession.DefaultRequestedMetadata, ISHType, returnFields, Enumerations.ActionMode.Read);
+                    string xmlIshObjects = IshSession.DocumentObj25.RetrieveMetadataByIshLngRefs(lngCardIds.ToArray(), requestedMetadata.ToXml());
+                    IshObjects retrievedObjects = new IshObjects(ISHType, xmlIshObjects);
+                    returnedObjects.AddRange(retrievedObjects.Objects);
+                }
+                else
+                {
+                    // ParameterGroups FilePath, FileContent, Metadata
+                    string resolution = Resolution ?? "";
+                    IshData ishData = null;
+                    var metadata = IshSession.IshTypeFieldSetup.ToIshMetadataFields(ISHType, new IshFields(Metadata), Enumerations.ActionMode.Update);
+                    string version = "-1";
+
+                    if (ParameterSetName == "ParameterGroupFilePath")
                     {
-                        string resolution = Resolution ?? "";
-                        var metadata = IshSession.IshTypeFieldSetup.ToIshMetadataFields(ISHType, new IshFields(Metadata), Enumerations.ActionMode.Update);
+                        ishData = new IshData(_edt, FilePath);
+                    }
 
-                        string version = "-1";
-                        if (Edt != null && FilePath != null)
+                    if (ParameterSetName == "ParameterGroupFileContent")
+                    {
+                        if (!_edt.Equals("EDTXML", StringComparison.Ordinal))
                         {
-                            IshData ishData = new IshData(Edt, FilePath);
-                            WriteDebug($"Id[{LogicalId}] Version[{Version}] Lng[{Lng}] Resolution[{resolution}] Metadata.length[{metadata.ToXml().Length}] dataSize[{ishData.Size()}]");
-                            DocumentObj25ServiceReference.UpdateResponse response = null;
-                            if (ShouldProcess(LogicalId + "=" + Version + "=" + Lng + "=" + resolution))
-                            {
-                                response = IshSession.DocumentObj25.Update(new UpdateRequest(
-                                    LogicalId,
-                                    Version,
-                                    Lng,
-                                    resolution,
-                                    metadata.ToXml(),
-                                    requiredCurrentMetadata.ToXml(),
-                                    ishData.Edt,
-                                    ishData.ByteArray));
-                                version = response.version;
-                            }
+                            throw new NotImplementedException("FileContent parameter is only supported with Edt='EDTXML'");
```

Comment:
This restriction is probably because -FileContent being a string, but not that in the parametersetgroup you do allow -Edt to be set (but only to value EDTXML then?!) ... so remove in the parameter, and document under -FileCOntent that EDT is hardcoded

Reply:
I thought I made it similar to Add-IshDocumentObj, where you still have Edt non-mandatory together with FileContent parameter, but it only works when set to EDTXML